packages/revocation/src/implementations/statuslistentry-verification.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  CredentialSubject,
+  StatusList2021Entry,
+  VerifiableCredential,
+} from '@ew-did-registry/credentials-interface';
+import * as pako from 'pako';
+import axios from 'axios';
+
+/**
+ * A Class to verify the StatusList2021Entry
+ */
+export class StatusListEntryVerification {
+  private verify: (vc: string, proof_options: string) => Promise<any>;
+
+  constructor(
+    callbackFunction: (vc: string, proof_options: string) => Promise<any>
+  ) {
+    this.verify = callbackFunction;
+  }
+
+  /**
+   * Verifies the status of StatusListEntry
+   * @param statusListEntry which needs to be verified
+   */
+  async verifyStatusList(statusListEntry: StatusList2021Entry) {
```

Comment:
I think this documentation is a bit misleading. At this point, the function isn't verifying the status of a StatusListEntry. It is verifying whether or not the `StatusListCredential` is valid. One must also call `verifyEncodedListStatus` to check the entry status, right?
```suggestion
  /**
   * Verifies the status of StatusListEntry
   * @param statusListEntry which needs to be verified
   */
  async verifyStatusList(statusListEntry: StatusList2021Entry) {
```

Reply:
The name is also confusing. It is `verifyStatusList` but the param is `statusListEntry`. Should it be `verifyStatusListEntry`?

Reply:
> So the idea is that one should only provide StatusListEntry from a VC and the API should verify everything (encodedList, StatuslListCredential, statusPurpose and so on). The verifyStatusList (should be verifyStatusListEntry) would be the main API which uses other sub methods within to completely verify or perform status check.

This makes sense to me, but let me know if you see it differently